### ISSUE TITLE
Install fails due to `clap_derive` incompatibility

### ISSUE THREAD

**Issue by intgr** [pksunkara/cargo-up#11]
Trying to install this with the recommended command `cargo install cargo-up --features cli --no-default-features` fails with build errors.

Looks like cargo uses the latest `clap_derive` version, but `cargo-up` is not compatible with it.

```
% cargo install cargo-up --features cli --no-default-features
    Updating crates.io index
  Installing cargo-up v0.0.5
[...]
   Compiling clap_derive v3.0.0
[...]
   Compiling clap v3.0.0-beta.4
[...]
   Compiling cargo-up v0.0.5
```

As a **work-around**, it's possible to use locked dependency versions: `cargo install cargo-up --features cli --no-default-features --locked`

<details>
<summary>Full error output</summary>

```
error: cannot find derive macro `Clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:20:17
   |
20 | #[derive(Debug, Clap)]
   |                 ^^^^
   |
note: `Clap` is imported here, but it is only a trait, without a derive macro
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:8:27
   |
8  | use clap::{crate_version, Clap};
   |                           ^^^^

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:26:7
   |
26 |     #[clap(short, long)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:30:7
   |
30 |     #[clap(long, hidden = true, requires_all = &["name", "path", "lib-path"])]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:34:7
   |
34 |     #[clap(long, hidden = true, requires_all = &["name", "to-version", "lib-path"])]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:38:7
   |
38 |     #[clap(long, hidden = true, requires_all = &["path", "to-version", "lib-path"], conflicts_with_all = &["version"])]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:42:7
   |
42 |     #[clap(long, hidden = true, requires_all = &["path", "name", "to-version"])]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/dep.rs:46:7
   |
46 |     #[clap(long, hidden = true)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find derive macro `Clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:10:17
   |
10 | #[derive(Debug, Clap)]
   |                 ^^^^
   |
note: `Clap` is imported here, but it is only a trait, without a derive macro
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:2:5
   |
2  | use clap::Clap;
   |     ^^^^^^^^^^

error: cannot find derive macro `Clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:15:17
   |
15 | #[derive(Debug, Clap)]
   |                 ^^^^
   |
note: `Clap` is imported here, but it is only a trait, without a derive macro
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:2:5
   |
2  | use clap::Clap;
   |     ^^^^^^^^^^

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:16:3
   |
16 | #[clap(version)]
   |   ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:19:7
   |
19 |     #[clap(long, value_name = "path")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:22:7
   |
22 |     #[clap(subcommand)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find derive macro `Clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:26:17
   |
26 | #[derive(Debug, Clap)]
   |                 ^^^^
   |
note: `Clap` is imported here, but it is only a trait, without a derive macro
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:2:5
   |
2  | use clap::Clap;
   |     ^^^^^^^^^^

error: cannot find attribute `clap` in this scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:27:3
   |
27 | #[clap(
   |   ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error[E0599]: no variant or associated item named `parse` found for enum `Cargo` in the current scope
  --> /home/marti/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-up-0.0.5/src/cli/main.rs:37:33
   |
32 | enum Cargo {
   | ---------- variant or associated item `parse` not found here
...
37 |     let Cargo::Up(opt) = Cargo::parse();
   |                                 ^^^^^ variant or associated item not found in `Cargo`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `parse`, perhaps you need to implement it:
           candidate #1: `Clap`

For more information about this error, try `rustc --explain E0599`.
error: failed to compile `cargo-up v0.0.5`, intermediate artifacts can be found at `/tmp/cargo-installo2axmU`

Caused by:
  could not compile `cargo-up` due to 15 previous errors
```
</details>

**Comment by ralyodio** [comment]
I get same issue. Is this project dead?

**Comment by pksunkara** [comment]
Well, the cli has never been fully done so I wasn't focusing on making it available until the bugs are fixed. I have a local branch that does this and it's still WIP.

The main offering of this repo right now is the lib. The cli is only used to test it locally for now.

But since you guys are curious, I corrected it up and published `0.0.6`.
